src/sagemaker/processing.py

```diff
@@ -389,25 +390,59 @@ def run(
         if wait:
             self.latest_job.wait(logs=logs)
 
-    def _get_user_script_name(self, code):
-        """Finds the user script name using the provided code file,
-        directory, or script name.
+    def _get_user_code_name(self, code):
+        """Gets the basename of the user's code from the URL the customer provided.
 
         Args:
-            code (str): This can be an S3 uri or a local path to either
-                a directory or a file.
+            code (str): A URL to the user's code.
+
+        Returns:
+            str: The basename of the user's code.
+
+        """
+        code_url = urlparse(code)
+        return os.path.basename(code_url.path)
+
+    def _handle_user_code_url(self, code):
```

Comment:
Did you mean to also validate that the s3 path is a directory and not a file? As it stands that validation only occurs for `elif code_url.scheme == "" or code_url.scheme == "file":`

Reply:
> Did you mean to also validate that the s3 path is a directory and not a file 

Nope, keeping the validation just to the "local" case is intentional. We'd have to make a service call to S3 to validate this in the S3 case, since the S3 path is a prefix to a collection of objects, and the file / directory distinction doesn't exist in S3.